Source/OpalGraphics/CGPath.m

```diff
@@ -311,9 +592,87 @@ void CGPathAddArcToPoint(
   CGFloat y1,
   CGFloat x2,
   CGFloat y2,
-  CGFloat r)
+  CGFloat radius)
 {
-  // FIXME:
+  CGFloat dx1, dy1, dx2, dy2;
+  CGFloat l, a1, a2;
+  CGPoint p;
+
+  p = CGPathGetCurrentPoint(path);
+ 
+  dx1 = p.x - x1;
+  dy1 = p.y - y1;
+  
+  l= dx1*dx1 + dy1*dy1;
+  if (l <= 0)

```

Comment:
Sorry, I had another look at your current code and the decision in CGPathAddArc now looks correct to me. And the logic in ArcToPoint has to be different. Forget this whole comment, I should have looked at the code of the whole file not just the changes.